skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -725,42 +668,49 @@ def _fast_nl_means_denoising_3d(image, int s=5, int d=7, double h=0.1,
         for t_pln in range(-d, d + 1):
             pln_dist_min = max(offset, offset - t_pln)
             pln_dist_max = min(n_pln - offset, n_pln - offset - t_pln)
+                    # alpha is to account for patches on the same column
+                    # distance is computed twice in this case
+            if t_pln == 0:
+                alpha = 1.0
+            else:
+                alpha = 0.5
             # Iterate over shifts along the row axis
             for t_row in range(-d, d + 1):
                 row_dist_min = max(offset, offset - t_row)
                 row_dist_max = min(n_row - offset, n_row - offset - t_row)
+                if t_row == 0:
+                    alpha = 1.0
+                else:
+                    alpha = 0.5
                 # Iterate over shifts along the column axis
                 for t_col in range(0, d + 1):
-                    col_dist_min = max(offset, offset - t_col)
-                    col_dist_max = min(n_col - offset, n_col - offset - t_col)
-                    # alpha is to account for patches on the same column
-                    # distance is computed twice in this case
-                    if t_col == 0 and (t_pln is not 0 or t_row is not 0):
-                        alpha = 0.5
-                    else:
-                        alpha = 1.0
+                    col_dist_min = offset
+                    col_dist_max = n_col - offset - t_col
 
                     # Compute integral image of the squared difference between
                     # padded and the same image shifted by (t_pln, t_row, t_col)
-                    integral[:, :] = 0
```

Comment:
Not sure why this only had [:, :] and not [:, :, :], but I guess either would be equivalent. Seems okay to remove it, but see comment above regarding doing a manual check of the demo output.